proto/internal/temporal/server/api/matchingservice/v1/request_response.proto

```diff
@@ -120,10 +121,13 @@ message AddWorkflowTaskRequest {
     // for TaskVersionDirective, which is unversioned.)
     temporal.server.api.taskqueue.v1.TaskVersionDirective version_directive = 10;
     temporal.server.api.taskqueue.v1.TaskForwardInfo forward_info = 11;
+    // Presence of this field means the workflow is versioned.
+    temporal.server.api.deployment.v1.WorkflowVersioningInfo workflow_versioning_info = 12;
```

Comment:
```suggestion
    temporal.server.api.deployment.v1.TargetWorkerDeployment target_worker_deployment = 12;
```

Reply:
Rationale: when History adds task/query to matching it expect it to be executed on this pinned or unpinned worker deployment. I would also remove word "worker" but then it can be confused with "deployment we are about to rollout". If you think it is not a case, you can call it just `target_deployment`.

Reply:
> pinned or unpinned worker deployment

deployment is not pinned or unpinned itself, workflow could be pinned or unpinned.

The idea is not that history gives matching the "target" deployment. It just gives matching the workflow versioning info and matching decides how to route based on that info. "Target deployment" does not fit well with unpinned because when matching does the redirect the deployment sent here would be the source of the redirect which will be confusing because "target" is almost antonym of "source".

Reply:
Conclusion: add the two fields to TaskVersionDirective